### ISSUE TITLE
Canonical redirects changed in recent WordPress

### ISSUE THREAD

**Issue by johngodley** [johngodley/redirection#949]
The way WordPress canonical redirects work have changed in a recent version of WordPress, and Redirection is no longer detecting them. This is causing a redirected URL for a previously redirected post to appear in the 404 log.

**Comment by cryptowally** [comment]
Looks like this is same thing I'm having - I use the AffiliateWP plugin which uses a /ref/blahblah format, and I have some redirects defined with your plugin that redirect to that landing page.

Since the upgrade to 4.9.6, I frequently get these 404 failures on pages that use the referral link.

```
Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Action: redirection/v1/redirect/?orderby=id&per_page=50 GET
Code: 404 Not Found
```
... frequently constantly on the site when redirects are used.  It doesn't do it with all redirects but it seems it does it with many many of them.

Full sanitized error is below.

Is there a workaround?  Since most of my affiliates and sales come in through these links, and now they have 404's, it means vastly reduced sales.

```
Plugin: 3.2
WordPress: 4.9.6 (single)
PHP: 7.0.29
Browser: Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0
JavaScript: https://mysite.com/wp-content/plugins/redirection/redirection.js
REST API: https://mysite.com/wp-json/

Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Action: redirection/v1/redirect/?orderby=id&per_page=50 GET
Code: 404 Not Found
Raw: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
   <head>
      <title>Error 404 - Not found</title>
      <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
      <meta http-equiv="cache-control" content="no-cache" />
   </head>
   <body style="font-family:arial;">
  	<h1 style="color:#0a328c;font-size:1.0em;">Error 404 - Not found</h1>
	<p style="font-size:0.8em;">Your browser can't find the document corresponding to the URL you typed in.</p>
   </body>
  </html>


Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Action: redirection/v1/group/?per_page=50 GET
Code: 404 Not Found
Raw: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
   <head>
      <title>Error 404 - Not found</title>
      <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
      <meta http-equiv="cache-control" content="no-cache" />
   </head>
   <body style="font-family:arial;">
  	<h1 style="color:#0a328c;font-size:1.0em;">Error 404 - Not found</h1>
	<p style="font-size:0.8em;">Your browser can't find the document corresponding to the URL you typed in.</p>
   </body>
  </html>
```

**Comment by johngodley** [comment]
That error does not look related. Can you post it in a new issue?

**Comment by cryptowally** [comment]
You're right - I had to do the old tried and true "save changes on
Permalinks page" fix and it magically started working again.

**Comment by johngodley** [comment]
Fixed in 3.3
